custom-properties/set_org_custom_properties.py

```diff
@@ -0,0 +1,152 @@
+        "allowed_values": None,
+        "values_editable_by": "org_and_repo_actors"
+    }
+    response = requests.put(url, headers=headers, json=data, verify=True)
+    if response.status_code != 200:
+        logging.error(f"Failed to define custom property for {org_name}: {response.json().get('message', 'Unknown error')}")
+    response.raise_for_status()
+    return response.status_code
+
+def get_org_repos(org_name):
+    """
+    Get all repositories in the organization.
+    """
+    url = f"{API_BASE}/orgs/{org_name}/repos"
+    response = requests.get(url, headers=headers, verify=True)
+    response.raise_for_status()
+    return response.json()
+
+def set_custom_properties(repo_full_name, properties):
+    """
+    Set custom properties for a repository.
+    """
+    owner, repo = repo_full_name.split('/')
+    url = f"{API_BASE}/repos/{owner}/{repo}/properties/values"
+    data = {
+        "properties": [
+            {"property_name": key, "value": value}
+            for key, value in properties.items()
+        ]
+    }
+    response = requests.patch(url, headers=headers, json=data, verify=True)
+    if response.status_code != 204:
+        logging.error(f"Failed to set properties for {repo_full_name}: {response.json().get('message', 'Unknown error')}")
+    response.raise_for_status()
+    return response.status_code
+
+def get_custom_properties(repo_full_name):
+    """
+    Get custom properties for a repository.
+    """
+    owner, repo = repo_full_name.split('/')
+    url = f"{API_BASE}/repos/{owner}/{repo}/properties/values"
+    response = requests.get(url, headers=headers, verify=True)
+    response.raise_for_status()
+    return response.json()
+
+def load_production_repos():
+    """
+    Load production repositories from production-repos.json file.
+    """
+    json_file_path = '../production-repos.json'
+    
+    try:
+        with open(json_file_path, 'r') as f:
+            repos = json.load(f)
+            if not isinstance(repos, list):
+                raise ValueError("JSON content is not a list")
+            return repos
+    except FileNotFoundError:
+        logging.error(f"Error: 'production-repos.json' not found at {os.path.abspath(json_file_path)}")
+        logging.error("Current working directory: %s", os.getcwd())
+        logging.error("Contents of the current directory: %s", os.listdir('.'))
+        raise
+    except json.JSONDecodeError as e:
+        logging.error(f"Error decoding JSON from {json_file_path}: {e}")
+        raise
+    except Exception as e:
+        logging.error(f"Unexpected error reading {json_file_path}: {e}")
+        raise
+
+# Define the custom property at the organization level
+try:
+    status = define_custom_property(ORG_NAME)
+    logging.info(f"Defined custom property for {ORG_NAME}: Status {status}")
+except requests.RequestException as e:
+    logging.error(f"Failed to define custom property for {ORG_NAME}: {str(e)}")
+
+# Get all repositories in the organization
```

Comment:
whats this doing? as per above:
Not sure why we want all repos in org, if we're passing in a list of accepted repos?

Reply:
Old config cleared now